odd_collector_azure/adapters/azure_sql/repository.py

```diff
@@ -0,0 +1,176 @@
+import textwrap
+from typing import Dict, Iterable, List
+
+import pytz
+import pyodbc
+
+from .domain import Dependency, DependencyType
+from .domain import Table, View
+
+
+TABLES_METADATA_QUERY = """
+    select
+        t.table_catalog,
+        t.table_schema,
+        t.table_name,
+        t.table_type,
+        s.create_date,
+        s.modify_date,
+        s.type_desc,
+        ps.row_count
+    from information_schema.tables t
+        left join sys.objects s
+            on
+                t.TABLE_NAME = s.NAME
+        left join sys.dm_db_partition_stats ps
+            on
+                s.object_id = ps.object_id and
+                s.type_desc = 'USER_TABLE'
+    where t.table_catalog = '{database}' and t.table_type = 'BASE TABLE'
+"""
+
+COLUMNS_METADATA_QUERY = """
+    select
+        c.table_catalog,
+        c.table_schema,
+        c.table_name,
+        c.column_name,
+        c.ordinal_position,
+        c.column_default,
+        c.is_nullable,
+        c.data_type,
+        c.character_maximum_length,
+        c.character_octet_length,
+        c.numeric_precision,
+        c.numeric_scale,
+        c.datetime_precision,
+        c.character_set_name,
+        c.collation_name,
+        t.column_name as column_key
+    from information_schema.columns c
+        left join INFORMATION_SCHEMA.CONSTRAINT_COLUMN_USAGE t
+            on
+                c.table_name = t.table_name and
+                c.column_name = t.column_name
+    where c.table_schema <> 'sys' and c.table_name = '{table_name}'
+"""
+
+VIEW_METADATA_QUERY = """
+    select
+        table_catalog,
+        table_schema,
+        table_name,
+        VIEW_DEFINITION
+    from information_schema.views
+    where table_schema <> 'sys'
+"""
+
+VIEW_DEPENDENCIES_QUERY = """
+    select
+        v.name,
+        dep.referenced_entity_name,
+        o.type as referenced_type
+    from sys.sql_expression_dependencies dep
+        left join sys.views v
+            on dep.referencing_id = v.object_id
+        left join sys.objects o
+            on dep.referenced_id = o.object_id
+"""
+
+
+class AzureSQLRepository:
+    def __init__(self, config):
+        self.config = config
+        self.connection_string = textwrap.dedent(
+            '''
+            Driver={driver};
+            Server={server};
+            Database={database};
+            Uid={username};
+            Pwd={password};
+            Encrypt=yes;
+            TrustServerCertificate=no;
+            Connection Timeout=30;
```

Comment:
Do not use hardcoded variables for `Encrypt`, `TrustServerCertificate`, `Connection Timeout`,  . move them to config as optional fields with default values

Reply:
Added these args to the AzureSQLPlugin class with default value.
Updated config_examples/azure_sql.yaml
Updated creating of connection string in the repository.py